integration-tests/c-example/lib/loops.c

```diff
@@ -0,0 +1,104 @@
+#include "loops.h"
+#include <stdbool.h>
+
+int while_loop(int n) {
+    int i = 0;
+    while (i < n) {
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    }
+    return 0;
+}
+
+int while_loop_unsigned(unsigned int n) {
+    int i = 0;
+    while (i < n) {
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    }
+    return 0;
+}
+
+unsigned int while_loop_unsigned_correct(unsigned int n) {
+    unsigned int i = 0;
+    while (i < n) {
+        i = i + 1;
+        if (n % i == 37U)
+            return 1;
+        else if (i == 50U)
+            return 2U;
+    }
+    return 0U;
+}
+
+int do_while_loop(int n) {
+    int i = 0;
+    do {
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    } while (i < n);
+    return 0;
+}
+
+int do_while_loop_unsigned(unsigned int n) {
+    int i = 0;
+    do {
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    } while (i < n);
+    return 0;
+}
+
+unsigned int do_while_loop_unsigned_correct(unsigned int n) {
```

Comment:
if you decide to leave this function, add it to header, please.

Reply:
done